prow/cmd/branchprotector/protect.go

```diff
@@ -63,45 +62,22 @@ func gatherOptions() options {
 	return o
 }
 
-func jobRequirements(jobs []config.Presubmit, after bool) []string {
-	var required []string
-	for _, j := range jobs {
-		// Does this job require a context or have kids that might need one?
-		if !after && !j.AlwaysRun && j.RunIfChanged == "" {
-			continue // No
-		}
-		if !j.SkipReport { // This job needs a context
-			required = append(required, j.Context)
-		}
-		// Check which children require contexts
-		required = append(required, jobRequirements(j.RunAfterSuccess, true)...)
-	}
-	return required
-}
-
-func repoRequirements(org, repo string, cfg config.Config) []string {
-	p, ok := cfg.Presubmits[org+"/"+repo]
-	if !ok {
-		return nil
-	}
-	return jobRequirements(p, false)
+type ref struct {
+	org, repo, branch string
 }
 
-type Requirements struct {
-	Org      string
-	Repo     string
-	Branch   string
-	Protect  bool
-	Contexts []string
-	Pushers  []string
+type requirements struct {
```

Comment:
I agree with this sort of thing generally (I had done something similar in my PR), but given that there is already a lot going on in this PR please keep as many lines the same as possible (I removed those changes from my PR to help simplify review).

Reply:
ok will fix